Able to add comments and retrieve comments

fnoori opened the pull request:
* When adding comments, first checks if a meme with that id exists
* When retrieving comments, expects a meme_id

Commit:
Able to add comments and retrieve comments

When adding comments, first checks if a meme with that id exists

When retrieving comments, expects a meme_id

fnoori commented:
https://trello.com/c/hHBYCF5l/21-create-handlers-for-meme-page-with-comments-with-realtime-broadcast-of-comments

adnanrh commented on src/server/main.js:
```diff
@@ -83,6 +83,22 @@ io.on('connect', (socket) => {
+
+    const commentResult = await meme.addComment(data, socketUser);
+
+    return socket.emit('addComment', commentResult);
```
Shouldn't this be io.emit, so it can be broadcasted to all users viewing the meme?

fnoori replied:
Yes it is ! Good call.

fnoori replied:
Wait no, this event only adds a comment, it returns if commenting was successful or not.

nxtman123 replied:
It adds a comment and if it was successful, it should broadcast into a room for those who are viewing the meme at the moment, as well as tell the commenter it was successful. There will need to be other events for entering and leaving the rooms. One room per meme.

nxtman123 replied:
Joining a room could be part of getMemeComments

fnoori replied:
@nxtman123 I'm not quite sure by what you mean by having events for entering/leaving rooms.
Could you explain that some more?

fnoori replied:
Do you mean like having an event for viewing the memes?

ie. the data being passed to the event would be the meme_id, then it would return the contents of the meme associated with that meme_id?

nxtman123 replied:
yup, that would work well. that handler would call `socket.join('meme-42') to join the room while it's returning the meme details. Another event would be needed for when the client navigates away from the meme page to call `socket.leave('meme-42')`.

Also, getting the existing memes (not the ones that are added while on the page) should have pagination, so that only the first (idk, 30) comments are loaded at first and then more can be loaded later (using the infinite scroll on the comments).

fnoori replied:
Alright, that's more clear.

So in this pull request: 
* I will modify the function so that it pushes the new comments to all the users viewing the certain meme and
* Modify the retrieve comments function so that it only retrieves the latest 30 comments

Commit:
lazy loading for comments now works

new requirements:

limit - how many comments to load
offset - how many to skip (frontend will need to change this each time new comments need to be loaded)

Commit:
Live commenting should work now

When user retrieves a single meme (with meme_id), they are added to a room

When a new comment is posted on a meme, that new comment is emitted to everyone viewing that particular meme (they are in the same room). This is a new event called "getLatestComment"

Commit:
fixed lint errors

adnanrh commented on src/server/meme.js:
```diff
@@ -18,4 +18,59 @@ module.exports = psql => ({
+      const comments = await psql('comments')
+        .where({ meme_id: data.meme_id })
+        .orderBy('date_created')
+        .limit(data.limit)
```
Do we want the limit to be sent in the request? Shouldn't it be a constant defined in meme.js?

fnoori replied:
Sure thing, I'll make it static

Commit:
Make limi of comment retrieval constant 20

Commit:
renamed functions
